### ISSUE TITLE
Brave Browser 1.1.23 in Ubuntu 19.1 Does Not Honor the Default Browser Flag

### ISSUE THREAD

**Issue by Firewall999** [brave/brave-browser#8894]
## Description 
Brave Browser 1.1.23 in Ubuntu 19.1 Does Not Honor the Default Browser Flag

## Steps to Reproduce

   1.   Set default browser flag in settings.
   2.   Quit the browser and reopen.

## Actual result:
Dialog asks user to make Brave the default Browser

## Expected result:
Should never see that damn dialog again!

## Reproduces how often: 
Every time.

## Brave version (brave://version info)

Brave	1.1.23 Chromium: 79.0.3945.88 (Official Build) (64-bit)
Revision	c2a58a36b9411c80829b4b154bfcab97e581f1f3-refs/branch-heads/3945@{#954}
OS	Linux
JavaScript	V8 7.9.317.32
Flash	(Disabled)
User Agent	Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36
Command Line	/snap/brave/63/opt/brave.com/brave/brave --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --enable-oop-rasterization=Enabled --sync-url=https://no-thanks.invalid --enable-features=PasswordImport,WebUIDarkMode,SimplifyHttpsIndicator --disable-features=AutofillServerCommunication,LookalikeUrlNavigationSuggestionsUI,NotificationTriggers,AudioServiceOutOfProcess,SmsReceiver,WebXR,WebXrGamepadModule,UnifiedConsent,AllowPopupsDuringPageUnload,SyncUSSBookmarks --flag-switches-begin --flag-switches-end --disable-webrtc-apm-in-audio-service
Executable Path	/snap/brave/63/opt/brave.com/brave/brave
Profile Path	/home/jackson/snap/brave/63/.config/BraveSoftware/Brave-Browser/Default

## Version/Channel Information:
  UNKNOWN

- Can you reproduce this issue with the current release?  UNKNOWN (UNABLE TO UPGRADE.  SEE BUG #8891)
- Can you reproduce this issue with the beta channel?  UNKNOWN
- Can you reproduce this issue with the dev channel?  UNKNOWN
- Can you reproduce this issue with the nightly channel?  UNKNOWN

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?  NO
- Does the issue resolve itself when disabling Brave Rewards?  NO
- Is the issue reproducible on the latest version of Chrome?  UNKNOWN

**Comment by kjozwiak** [comment]
@Firewall999 assuming you're using a version from Snap as per https://github.com/brave/brave-browser/issues/8891? `1.1.23` is a pretty old version. Maybe download the latest version from `apt` as @bsclifton recommended via https://github.com/brave/brave-browser/issues/8891#issuecomment-605209172? This way you can grab the latest version and see if the `Set Brave as Default` is working in that version.

**Comment by Firewall999** [comment]
Thank you for responding!

I am an Ubuntu neophyte. Don't know snap from a hole it the wall.

I think I did use apt-get and apt install. I definitely saw something like "1.5.xx installed" in terminal.

Software Updater also successfully downloaded a new Brave version, but it disappeared into the ether and did not install.  Was that via snap?

If you give me the appropriate terminal commands, I can use them in terminal.  I can then copy them and include them in bug report updates.

Thank you for your time.

Firewall999

PS.  Part of the problem is that there is no help online for upgrading apps, only for installing them.

**Comment by kjozwiak** [comment]
Does the issue resolve it's self if you run through https://brave-browser.readthedocs.io/en/latest/installing-brave.html#linux again and try updating/installing? It won't touch your profile so you won't lose any data.

@fmarier any idea why this could be happening if the above doesn't work?

**Comment by Firewall999** [comment]
ISSUE NOT RESOLVED

Here's what I got while installing (reinstalling?) Brave:
sudo apt install apt-transport-https curl
[sudo] password for jackson: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
curl is already the newest version (7.65.3-1ubuntu3).
apt-transport-https is already the newest version (1.9.4).
0 upgraded, 0 newly installed, 0 to remove and 8 not upgraded.
jackson@stonewall:~$ curl -s https://brave-browser-apt-release.s3.brave.com/brave-core.asc | sudo apt-key --keyring /etc/apt/trusted.gpg.d/brave-browser-release.gpg add -
OK
jackson@stonewall:~$ echo "deb [arch=amd64] https://brave-browser-apt-release.s3.brave.com/ stable main" | sudo tee /etc/apt/sources.list.d/brave-browser-release.list
deb [arch=amd64] https://brave-browser-apt-release.s3.brave.com/ stable main
jackson@stonewall:~$ sudo apt update
Hit:1 http://us.archive.ubuntu.com/ubuntu eoan InRelease
Get:2 http://us.archive.ubuntu.com/ubuntu eoan-updates InRelease [97.5 kB]                                                      
Get:3 http://security.ubuntu.com/ubuntu eoan-security InRelease [97.5 kB]                                                       
Ign:4 http://dl.google.com/linux/earth/deb stable InRelease                                                                     
Hit:5 https://brave-browser-apt-release.s3.brave.com stable InRelease                                                           
Get:6 http://dl.google.com/linux/earth/deb stable Release [933 B]                                                               
Get:7 http://us.archive.ubuntu.com/ubuntu eoan-backports InRelease [88.8 kB]
Get:8 http://dl.google.com/linux/earth/deb stable Release.gpg [819 B]
Get:9 http://us.archive.ubuntu.com/ubuntu eoan-updates/main amd64 Packages [244 kB]
Get:10 http://us.archive.ubuntu.com/ubuntu eoan-updates/main i386 Packages [189 kB]
Get:11 http://security.ubuntu.com/ubuntu eoan-security/main amd64 DEP-11 Metadata [7,392 B]
Get:12 http://security.ubuntu.com/ubuntu eoan-security/main amd64 c-n-f Metadata [4,836 B]
Get:13 http://security.ubuntu.com/ubuntu eoan-security/universe amd64 DEP-11 Metadata [4,496 B]
Get:14 http://security.ubuntu.com/ubuntu eoan-security/universe DEP-11 48x48 Icons [7,591 B]            
Get:15 http://security.ubuntu.com/ubuntu eoan-security/universe DEP-11 64x64 Icons [9,785 B]           
Get:16 http://us.archive.ubuntu.com/ubuntu eoan-updates/main amd64 DEP-11 Metadata [78.9 kB]   
Get:17 http://us.archive.ubuntu.com/ubuntu eoan-updates/main DEP-11 48x48 Icons [14.6 kB]                
Get:18 http://us.archive.ubuntu.com/ubuntu eoan-updates/main DEP-11 64x64 Icons [24.7 kB]            
Ign:8 http://dl.google.com/linux/earth/deb stable Release.gpg                                    
Get:19 http://us.archive.ubuntu.com/ubuntu eoan-updates/universe i386 Packages [195 kB]              
Get:20 http://security.ubuntu.com/ubuntu eoan-security/universe amd64 c-n-f Metadata [6,116 B]
Get:21 http://us.archive.ubuntu.com/ubuntu eoan-updates/universe amd64 Packages [201 kB]                  
Get:22 http://us.archive.ubuntu.com/ubuntu eoan-updates/universe Translation-en [73.7 kB]                 
Get:23 http://us.archive.ubuntu.com/ubuntu eoan-updates/universe amd64 DEP-11 Metadata [30.1 kB]
Get:24 http://us.archive.ubuntu.com/ubuntu eoan-updates/universe DEP-11 48x48 Icons [19.0 kB]
Get:25 http://us.archive.ubuntu.com/ubuntu eoan-updates/universe DEP-11 64x64 Icons [36.9 kB]
Get:26 http://us.archive.ubuntu.com/ubuntu eoan-backports/universe amd64 DEP-11 Metadata [7,760 B]
Reading package lists... Done                                              
W: GPG error: http://dl.google.com/linux/earth/deb stable Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 78BD65473CB3BD13
E: The repository 'http://dl.google.com/linux/earth/deb stable Release' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
jackson@stonewall:~$ sudo apt install brave-browser
Reading package lists... Done
Building dependency tree       
Reading state information... Done
**brave-browser is already the newest version (1.5.115).**
0 upgraded, 0 newly installed, 0 to remove and 8 not upgraded.
jackson@stonewall:~$ 

(Not sure why so much is crossed out in preview.  It was not so in terminal.)  Please notice the third to last line.  [Bold added by author.]

x-special/nautilus-clipboard
copy
file:///home/jackson/Pictures/Screenshot%20from%202020-03-30%2012-23-38.png

Thank you for your time.

**Comment by fmarier** [comment]
I would start by uninstalling the snap as I suggested in https://github.com/brave/brave-browser/issues/8891#issuecomment-606137174 to ensure there's only one version of Brave on your system.

Once you've done that, we can look into why the remaining version of Brave is not set as the default, if that's still an issue.

With respect to the weird crossed-out lines in your output, this is due to the way that GitHub interprets various symbols as formatting instructions. If you want to paste text undisturbed from these things, you need to do it in between three slanted quote marks (```):

![Screenshot from 2020-03-30 10-39-54](https://user-images.githubusercontent.com/167821/77943867-ced31680-7272-11ea-8ea2-e773b740bc81.png)

I've edited your last message to add the quote marks, which removed all of the unwanted formatting.

**Comment by Firewall999** [comment]
I tried to uninstall snap, and this is what I got:

'''sudo snap remove brave
[sudo] password for jackson: 
error: cannot perform the following tasks:
- Save data of snap "brave" in automatic snapshot set #1 (cannot create archive: tar: 63/.config/BraveSoftware/Brave-Browser/Default/publisher_info_db-journal: file changed as we read it (and 1 more))'''

It is not at all clear from this error message whether the command completed or not.

Nothing has changed, so I guess it did not work.

**Comment by fmarier** [comment]
> I tried to uninstall snap, and this is what I got:
> ```
> sudo snap remove brave
> [sudo] password for jackson:
> error: cannot perform the following tasks:
> 
> * Save data of snap "brave" in automatic snapshot set #1 (cannot create archive: tar: 63/.config/BraveSoftware/Brave-Browser/Default/publisher_info_db-journal: file changed as we read it (and 1 more))'''
> ```

Try the following:

   sudo snap remove --purge brave

**Comment by Firewall999** [comment]
2 PROBLEMS SOLVED BY REMOVING DUPLICATE INSTALL

1. Now running latest version
2. Default flag now being honored.

THANK YOU!
     On Monday, March 30, 2020, 07:55:06 PM EDT, Francois Marier <notifications@github.com> wrote:  
 
 



I tried to uninstall snap, and this is what I got:
sudo snap remove brave[sudo] password for jackson:error: cannot perform the following tasks:* Save data of snap "brave" in automatic snapshot set #1 (cannot create archive: tar: 63/.config/BraveSoftware/Brave-Browser/Default/publisher_info_db-journal: file changed as we read it (and 1 more))'''

Try the following:

sudo snap remove --purge brave
